src/Serializer/DoctrineDenormalizer.php

```diff
@@ -126,16 +132,21 @@ private function denormalizeNonArray($data, string $class, ?array $context = nul
         if (\is_string($data) !== true && \is_int($data) !== true) {
             return $data;
         }
+
         $keys = $this->getClassLookupKey($class);
-        // If there's more than a single key, and the defined default, we've got a composite key, which we can't
-        // handle with a single scalar value.
+
+        // If there's more than a single key, and the defined default, we've got a
+        // composite key, which we can't handle with a single scalar value.
```

Comment:
Shouldn't we do this _before_ we do the entity lookup?

Reply:
It does happen before the lookup (which happens on L145)

Reply:
oops, missed the closing method `}` nevermind